### DISCUSSION TITLE
How to disable "Token OTP" in command line?

### DISCUSSION THREAD

**Discussion by gokubr** [ngoduykhanh/PowerDNS-Admin#1506] [Q&A] [answered]
I need urgent help to access via Power DNS graphical interface.
One of the admins mistakenly activated the "OTP Token", but we weren't supposed to use that option, and now we can't access the web system.
We have credentials to access the server and needed to find out how to disable this option on the server to have normalized access.

![0000](https://user-images.githubusercontent.com/26436542/229791618-ebe37dcb-e22f-4f1e-a7de-3a0b01e3badc.JPG)

**Comment by subbink** [comment, marked as answer]
There is a `otp_force` setting in the setting table of PowerDNS-Admin, which you could set to false.

**Reply by gokubr** [reply]
Hello, thanks for the answer, but I've been looking all morning and I haven't found the file to make this change and test. Can you help me?

**Reply by AzorianMatt** [reply]
@gokubr as another person said, this is not a file but the associated database. It technically would be a single file if you have your instance configured to use SQLite instead of MySQL or PGSQL.

**Comment by arbaldin** [comment]
Hi @gokubr,

It's not a file, you should query the database itself and change the value otp_force in the setting table on the PowerDNS-Admin database.

Regards.

**Comment by gokubr** [comment]
@arbaldin @AzorianMatt  @subbink 

Hello everyone, thank you very much for the answers, I have no experience with databases and this was a great help.

I added the "otp_force" field in the "setting" table, but I still don't get the access, the Token is still requested. I don't know what else to do in this case.

![power-dns](https://user-images.githubusercontent.com/26436542/230972554-304bba7a-2b19-4de3-b23e-447319469c41.JPG)

**Reply by AzorianMatt** [reply]
@gokubr you said you added the setting. If you just created a new row with the expected values, I'm guessing you have taken the wrong approach. When you enabled that setting in the first place, a row should have already been created in the settings table for that setting. You need to update that row, as the one you added is likely being ignored in favor of the original row.

**Reply by gokubr** [reply]
That's right. I also found it strange, but before that line did not exist and the configuration table only has this information that I sent in the print, no more lines.

**Reply by AzorianMatt** [reply]
@gokubr forgive me as I don't mean to sound rude, but there must be an oversight here. If you did indeed enable the global OTP requirement setting, it most certainly has to be in that table for it to take effect. There is nowhere else in the application that this is stored.

**Reply by gokubr** [reply]
I understand, not rude at all. On the contrary, I really appreciate everyone's help.
It's just that I really can't perform the login because the OTP Token is requested, so it's activated, but for some reason it's not in this configuration table. 

I'll check the settings again, maybe I missed something.

**Reply by AzorianMatt** [reply]
@gokubr certainly happy to help. Are you absolutely certain that the global setting is what was enabled? Any chance that you just enabled it for your individual user? That would be stored somewhere else in that scenario.

**Reply by gokubr** [reply]
It was another sysadmin who made this change.
If this is what you mentioned, which database table would you be storing it in?

**Reply by AzorianMatt** [reply]
@gokubr I believe that would be stored directly on the user record in the user table.

**Reply by gokubr** [reply]
Unfortunately in the user table there is no "opt_force", only "otp_secret".
I'll keep looking for something, but I don't have any more ideas.

**Comment by elmobp** [comment]
Check for ```otp_field_enabled``` in the settings table?

**Reply by gokubr** [reply]
Hello @elmobp , this is the only information in the database setting table.

![image](https://user-images.githubusercontent.com/26436542/231154359-7f47c561-71a2-4dd8-a0b1-c095ecd89533.png)

**Reply by gokubr** [reply]
Field "otp_field_enabled" does not exist in this table.

**Comment by gokubr** [comment]
I keep trying to figure something out, but to no avail.
Does anyone have any other suggestions or ideas of what I can do?

**Reply by AzorianMatt** [reply]
@gokubr are you running this from a docker image? If so, what is the full repository path, image name, and tag you are using? I couldn't help but notice that your screenshot is definitely of an older release, and I'm wondering what release that is or if it's even from an official project image (as there are a number of third-party images out there which are outdated).

**Reply by gokubr** [reply]
Yes, it's through a Docker container, but it was inherited from the previous admin, I didn't install it.
It is this:


![image](https://user-images.githubusercontent.com/26436542/231861997-31e19e3a-a02c-40fa-bc81-983a4819cf0a.png)

**Reply by AzorianMatt** [reply]
@gokubr at least we're getting somewhere now. That is the old image repository and that is no longer actively updated.

Assuming it had been updating at least at some point, I would guess you had the latest version of `0.3.0` from there. You can get the same with this image:

`powerdnsadmin/pda-legacy:0.3.0`

OR you can try upgrading to the latest with:

`powerdnsadmin/pda-legacy:latest`

**Reply by gokubr** [reply]
I updated to the latest image as mentioned.
I checked the database tables and other settings and I can't login or create a new user to access.

![image](https://user-images.githubusercontent.com/26436542/231917667-ce9bec8f-921e-43bb-b779-7c592e7dcdd0.png)

**Reply by AzorianMatt** [reply]
@gokubr why don't you just wipe out the installation and start fresh? At that point, it could just re-import everything or you could manually re-import just some tables.

**Reply by gokubr** [reply]
I did that and it worked, I got access, now everything is ok.
I apologize for bothering you so much with so many questions, it's just that I didn't know the tool and the database well.

Thank you for your patience and help, your help was essential. Thank you very much,

![image](https://user-images.githubusercontent.com/26436542/231942412-67651d01-2643-41ae-9586-3c2663028eef.png)

**Reply by AzorianMatt** [reply]
@gokubr you're welcome and no worries!

**Comment by gokubr** [comment]
Problem solved.
Many thanks to everyone who helped in any way! 👍
